File: src/main/java/com/dambra/paul/moduluschecker/ModulusAlgorithm.java
package com.dambra.paul.moduluschecker;

public enum ModulusAlgorithm {
    MOD10, MOD11, DOUBLE_ALTERNATE;

    public static ModulusAlgorithm parse(String s) throws UnknownAlgorithmException {
        switch (s){
            case "MOD10": return MOD10;
            case "MOD11": return MOD11;
            case "DBLAL": return DOUBLE_ALTERNATE;
        }
        throw new UnknownAlgorithmException("Cannot parse " + s + " as a modulus algorithm");
    }
}

File: src/main/java/com/github/pauldambra/moduluschecker/ModulusAlgorithm.kt
package com.github.pauldambra.moduluschecker

enum class ModulusAlgorithm {
    MOD10, MOD11, DOUBLE_ALTERNATE;


    companion object {

        @Throws(UnknownAlgorithmException::class)
        fun parse(s: String): ModulusAlgorithm {
            when (s) {
                "MOD10" -> return MOD10
                "MOD11" -> return MOD11
                "DBLAL" -> return DOUBLE_ALTERNATE
                else    -> throw UnknownAlgorithmException("Cannot parse $s as a modulus algorithm")
            }
        }
    }
}
